core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -189,13 +190,12 @@ private[spark] class HighlyCompressedMapStatus private (
     emptyBlocks.readExternal(in)
     avgSize = in.readLong()
     val count = in.readInt()
-    val hugeBlockSizesArray = mutable.ArrayBuffer[Tuple2[Int, Byte]]()
+    hugeBlockSizes = new util.HashMap[Int, Byte](count).asScala
```

Comment:
How about just scala's mutable Map? I'd expect it's no slower than Java's, given it might specialize for primitives (not sure about this) and sometimes has smarter implementations internally.

Reply:
scala.mutable.HashMap implementation does not have a way to set initial capacity out of the box. The performance gets worse, probably because of resizing hash table. 

scala.mutable.OpenHashMap implementation does, but it is still slower than java.util.HashMap 

However, if  a kind of tradeoff between code cleanness and performance os needed, I would use one of the variants above.

Reply:
Does calling `HashMap.sizeHint(...)` here actually help?
I'd stick to the scala collection for now